src/rust/engine/src/nodes.rs

```diff
@@ -1054,26 +1056,46 @@ impl Node for NodeKey {
     };
 
     let context2 = context.clone();
-    future::lazy(|| {
-      if let Some(span_id) = maybe_span_id {
-        set_parent_id(span_id);
-      }
-      match self {
-        NodeKey::DigestFile(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::DownloadedFile(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::MultiPlatformExecuteProcess(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::ReadLink(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Scandir(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Select(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Snapshot(n) => n.run(context).map(NodeResult::from).to_boxed(),
-        NodeKey::Task(n) => n.run(context).map(NodeResult::from).to_boxed(),
+    future::lazy({
+      let context = context.clone();
+      let self2 = self.clone();
+      move || {
+        if let Some(path) = self2.fs_subject() {
```

Comment:
Relatively few nodes have `fs_subjects`, so when you fix this to spawn on the `io_pool`, you will definitely want to avoid spawning if the node doesn't.

Reply:
will remove the future::lazy here, as it isn't needed.